### PR TITLE
Initial implementation.

### PR DESCRIPTION
**Pull request by gannan08**

### CODE REVIEW

File: lib/sha.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * Hashes a string of data using SHA-256 or SHA-384.
+ *
+ * @param {object} options - The options to use.
+ * @param {string} options.algorithm - The algorithm to use.
+ * @param {string} options.string - The string to hash.
+ *
+ * @returns {Uint8Array} The hash digest.
+ */
+export async function sha({algorithm, string}) {
+  algorithm = algorithm === 'SHA-256' ? 'sha256' : 'sha384';
```
Comment (dlongley) [inline lib/sha.js]: Only `sha256` is ever used.

```suggestion
  algorithm = algorithm === 'SHA-256' ? 'sha256' : 'sha384';
```

**Review by dlongley — APPROVED** [review]
One minor nit. We don't need anything to say "sha-384" anywhere in this package.

File: lib/sha.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2024 Digital Bazaar, Inc. All rights reserved.
+ */
+import crypto from 'node:crypto';
+
+const REQUIRED_HASH_ALGORITHM = 'SHA-256';
+
+/**
+ * Hashes a string of data using SHA-256 or SHA-384.
```
Comment (davidlehn) [inline lib/sha.js]:
```suggestion
 * Hashes a string of data using SHA-256.
```

File: lib/sha-browser.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2024 Digital Bazaar, Inc. All rights reserved.
+ */
+/* eslint-env browser */
+const crypto = self && (self.crypto || self.msCrypto);
+
+const REQUIRED_HASH_ALGORITHM = 'SHA-256';
+
+/**
+ * Hashes a string of data using SHA-256 or SHA-384.
```
Comment (davidlehn) [inline lib/sha-browser.js]:
```suggestion
 * Hashes a string of data using SHA-256.
```

**Pull request merged**